src/Kestrel.Core/Internal/HttpConnection.cs

```diff
@@ -622,13 +641,11 @@ void IConnectionTimeoutFeature.ResetTimeout(TimeSpan timeSpan)
             ResetTimeout(timeSpan.Ticks, TimeoutAction.AbortConnection);
         }
 
-        private void CloseUninitializedConnection()
+        private void CloseUninitializedConnection(ConnectionAbortedException abortReason)
         {
             Debug.Assert(_adaptedTransport != null);
 
-            // CancelPendingRead signals the transport directly to close the connection
-            // without any potential interference from connection adapters.
-            _context.Application.Input.CancelPendingRead();
+            _context.ConnectionFeatures.Get<IConnectionLifetimeFeature>().Abort(abortReason);
```

Comment:
We should store the ConnectionContext on the `HttpConnectionContext`.

Reply:
Ugh, see https://github.com/aspnet/KestrelHttpServer/pull/2612/files#r191705652

Reply:
Seems reasonable. Might as well get it right if we're going to break things. I agree that aborting a connection is common enough to deserve a top-level method on the ConnectionContext. Interestingly enough this will be the first non-property on the ConnectionContext.

I think if we made the Abort(ConnectionAbortedException) method virtual instead of abstract on ConnectionContext, it would technically be non-breaking. Do you think we should do that?

Reply:
The risk of making Abort virtual is that transports might not implement it. On second thought, I would rather just break compilation.